.github/workflows/python-app.yml

```diff
@@ -7,59 +7,63 @@ on:
-        # by a year, and is currently incompatible with the format of this repo.
-        wget -O/tmp/Miniconda3-latest-Linux-x86_64.sh https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh
-        bash /tmp/Miniconda3-latest-Linux-x86_64.sh -b -f
-        source ~/miniconda3/bin/activate # careful! after this we're no longer in GH action's python, we're in conda's!
-        conda install -y -c conda-forge git-annex
-        # install proper NodeJS for bids-validator
-        sudo rm -rf ~/.nvm
-        curl -sL https://deb.nodesource.com/setup_current.x | sudo -E bash -
-        sudo apt-get install -y nodejs
-        sudo npm install -g bids-validator
-    - name: Include miniconda in PATH
+    - name: Install git-annex
       run: |
-        echo "$HOME/miniconda3/bin" >> $GITHUB_PATH
-        echo "$HOME/miniconda3/Scripts" >> $GITHUB_PATH
+        sudo apt-get update
+        sudo apt-get --yes install git-annex
+        # the user name and email is needed by git-annex to write to
+        # the local git-annex branch
         git config --global user.name "gh-actions"
         git config --global user.email "gh-actions"
-    - name: setup
+        # save space by using hard links to annexed files
+        git config --global annex.thin true
+        # retry for transient failures when downloading annexed file contents
+        git config --global annex.retry 2
+
+    - name: Download dataset
       run: |
-        # download the data content
-        git fetch origin git-annex:git-annex # Travis does a shallow checkout, so it is missing this important branch
-        git annex init
-    - name: download data
-      run: |        
-        echo -e "\e[1m \e[4m \e[34mDownloading dataset; this will take a while. \n"
-        # this is -q because git-annex doesn't have a short progress bar like git
-        # and the download log is overwhelming and not interesting and might trip over
-        #    https://docs.travis-ci.com/user/common-build-problems/#log-length-exceeded.
-        # but -q might instead trip over
-        #   https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
-        # so it's a balancing act.
-        git annex get -J8
-    - name: script
+        git fetch --depth=1 origin git-annex:git-annex
+        git annex get --jobs 8
+
+    - name: Install Deno
+      uses: denoland/setup-deno@v2
+
+    - name: Install Python 3.13
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.13
+
+    - name: Install spine-generic
+      run: |
+        pip install spinegeneric@git+https://github.com/spine-generic/spine-generic.git@v2.9
+
+    - name: Run bids-validator
+      # I would prefer to whitelist the allowed environment variables,
+      # but upstream adds new ones unpredictably. At least we don't
+      # allow any network access.
+      run: |
+        deno run --no-prompt --allow-env --allow-read=. jsr:@bids/validator .
```

Comment:
This is my first time seeing `deno` before! After some reading up on the options, the syntax looks alright to me, but out of curiosity, do the upstream `bids-validator` docs show example syntax? I'm curious if there's anything we can link to that demonstrates that this is the proper `deno` syntax to use for this project specifically.

Reply:
The IGNORE_TEST_WIN32 is super curious. I found https://github.com/kaelzhang/node-ignore/issues/152 but not much else? Hrm...